server/controllers/auth/signup.js

```diff
@@ -0,0 +1,83 @@
+// modules 
+const bcrypt = require('bcrypt');
+const uuid = require('uuid/v1');
+const jwt = require('jsonwebtoken');
+
+const database = require("../../database");
+
+exports.signup = (req, res, next) => {
+  const email = req.body.email;
+  // password cannot be longer than 72 characters
+  const password = req.body.password;
+
+  // validate member existance in database
+  database.query(`
```

Comment:
Maybe use tools such as https://knexjs.org/ for query building? It comes with migration options as well. Not sure how you are managing migrations right now.

Reply:
I am also not sure about the migration, I haven't look much into that.

But I did thought of using sequelize too, but using ORMs doesn't provide much flexibility as compare to native SQL language.

Reply:
> but using ORMs doesn't provide much flexibility as compare to native SQL language.

Technically true. But haven't come across any query that I couldn't write with ORM. Plus, you can always use regular SQL if such instance occurs. Overall, ORMs provide better readability.

Plus, the migration is an important factor. If it were a MongoDB, it would create the DB "schema" on its own. But SQL DBs don't. If multiple people were to work on this project, setup would become a hassle.